core/src/main/scala/zio/http/model/Header.scala

```diff
@@ -18,4 +18,102 @@
 
 package zio.http.model
 
-case class Header(name: String, value: String)
+import java.io.Serializable
+import java.net.{ URI, URL }
+import java.time.Instant
+
+import zio.duration.Duration
+import zio.http.model.ResponseHeader.ETag
+
+sealed trait RequestHeader extends Product with Serializable
+
+object RequestHeader {
+
+  final case class Accept(mediaRange: List[MediaRange]) extends RequestHeader
+
+  final case class AcceptCharset(charsets: List[CharsetRange]) extends RequestHeader
+
+  final case class AcceptEncoding(contentCodings: List[ContentCoding]) extends RequestHeader
+
+  final case class AcceptLanguage(languages: List[Language]) extends RequestHeader
+
+  final case class Authorization(scheme: AuthenticationScheme, credentials: Credentials) extends RequestHeader
+
+  final case class Expect(value: String) extends RequestHeader
+
+  final case class From(value: Email) extends RequestHeader
+
+  final case class Host(host: URL, port: Int) extends RequestHeader
+
+  final case class IfMatch(etags: List[ETag]) extends RequestHeader
+
+  final case class IfModifiedSince(value: Instant) extends RequestHeader
+
+  final case class IfNoneMatch(etags: List[ETag]) extends RequestHeader
+
+  final case class IfRange(value: Either[ETag, Instant]) extends RequestHeader
+
+  final case class IfUnmodifiedSince(value: Instant) extends RequestHeader
+
+  final case class MaxForwards(value: Int) extends RequestHeader
+
+  final case class ProxyAuthorization(scheme: AuthenticationScheme, credentials: Credentials) extends RequestHeader
+
+  final case class Range(value: String) extends RequestHeader
+
+  final case class Referer(value: URI) extends RequestHeader
+
+  final case class TransferEncoding(transferEncodings: List[TransferEncodingType]) extends RequestHeader
+
+  sealed trait TransferEncodingType extends Product with Serializable
+  case object CHUNKED               extends TransferEncodingType
+  case object IDENTITY              extends TransferEncodingType
+  case object GZIP                  extends TransferEncodingType
+  case object COMPRESS              extends TransferEncodingType
+  case object DEFLATE               extends TransferEncodingType
```

Comment:
```suggestion
object TransferEncodingType {
  case object CHUNKED               extends TransferEncodingType
  case object IDENTITY              extends TransferEncodingType
  case object GZIP                  extends TransferEncodingType
  case object COMPRESS              extends TransferEncodingType
  case object DEFLATE               extends TransferEncodingType 
}
```
I've noticed that some of the new code doesn't have case classes inside companion object of the related selaed trait

Reply:
Is it required?